src/raml.coffee

```diff
@@ -171,14 +172,39 @@ class @RamlParser
         return if lastVisitedNode then lastVisitedNode else node
       )
 
-  getPendingFile: (loader, fileInfo) ->
+  dereferenceSchema: (file) ->
+    deferred = @q.defer()
+    schemaLoader.dereference(file)
+      .then((schema) ->
+        deferred.resolve(JSON.stringify(schema))
+      ).catch((err) ->
+        deferred.reject(err)
+      )
+    return deferred.promise
+
+  checkPendingFileIsSchema: (loader, fileInfo, node) ->
+    if fileInfo.parentKey and fileInfo.parentKey.value is 'schema'
+      return true
+
+    if fileInfo.parentNode
+      schemasSeqNode = loader.findSchemasSequenceNode(node)
```

Comment:
What if you have this RAML:

```raml
#%RAML 0.8
/foo:
  post:
    body:
      schema: !include someFastFile.json
schemas: !include someSlowFile.yaml
```

You should defer all schema reading and processing, until all !included files have been loaded.